engine/queue/score-queue-impl/src/main/java/io/cloudslang/engine/queue/repositories/ExecutionQueueRepositoryImpl.java

```diff
@@ -320,17 +392,40 @@ public int getBatchSize() {
     }
 
     @Override
-    public List<ExecutionMessage> poll(String workerId, int maxSize, long workerPollingMemory, ExecStatus... statuses) {
+    public List<ExecutionMessage> poll(
+            String workerId, int maxSize, long workerPollingMemory, ExecStatus... statuses) {
+
+        Object[] args;
+        if (useLargeMessageQuery) {
+            args = preparePollArgs(workerId, workerPollingMemory, statuses);
+        } else {
+            args = prepareStdPollArgs(workerId, statuses);
+        }
 
-        pollJdbcTemplate.setStatementBatchSize(maxSize);
+        String sqlStat = workerQuery.replaceAll(":status", StringUtils.repeat("?", ",", statuses.length));
+
+        return executePoll(maxSize, sqlStat, args);
+    }
+
+    private String getLargeMessageQuery() {
+        return isMSSQL() ? QUERY_WORKER_SQL_MSSQL : QUERY_WORKER_SQL;
+    }
 
+    private boolean isMSSQL() {
```

Comment:
preffered name according to score best practices that follows google code convention is 

> isMssql()